crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -1284,12 +1673,63 @@ fn retain_names_from_declared_idents(child_names: &mut Vec<Name>, current_declar
     *child_names = retained_names;
 }
 
-fn gen_ident(cnt: &mut u32) -> JsWord {
+fn gen_action_ident(cnt: &mut u32) -> JsWord {
     let id: JsWord = format!("$$ACTION_{cnt}").into();
     *cnt += 1;
     id
 }
 
+fn gen_cache_ident(cnt: &mut u32) -> JsWord {
+    let id: JsWord = format!("$$CACHE_{cnt}").into();
```

Comment:
I think we also need to patch 
https://github.com/vercel/next.js/blob/af9b229074e39617bdd32d390fdc8f6a36d7bcde/turbopack/crates/turbopack-ecmascript/src/tree_shake/graph.rs#L391-L454

Reply:
We'll need to generalize these 👍